azure/scope/machinepool.go

```diff
@@ -454,6 +462,28 @@ func (m *MachinePoolScope) applyAzureMachinePoolMachines(ctx context.Context) er
 	return nil
 }
 
+// DeleteAllAzureMachinePoolMachines deletes all AzureMachinePoolMachines for the AzureMachinePool.
+func (m *MachinePoolScope) DeleteAllAzureMachinePoolMachines(ctx context.Context) error {
```

Comment:
Are the AzureMachinePoolMachines owned by the Cluster or MachinePool? if so, could we leverage [k8s garbage collection](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy) instead of deleting all the AMPM from here?

Reply:
The AzureMachinePoolMachines are owned by the AzureMachinePool as well as a Machine. That Machine is owned by the MachinePool. What would you recommend?